### PR TITLE
Interactive bots: Improve task descriptions

### PR DESCRIPTION
**Pull request by abhayKashyap03**

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@abhayKashyap03, it looks like you've already signed the Dropbox CLA. Thanks!

File: tasks/interactive-bots.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ Follow the tutorial below to test run the followup bot on your Zulip instance
 
 * Run the Zulip server and log in the browser as Cordelia.
 
-* Find the API keys.
+* Find the API keys (Settings -> Your bots -> Show/change your API).
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: We could put the info in brackets on line 46, before the first full stop.

File: tasks/interactive-bots.md [hunk 2]
```diff
@@ -54,19 +54,17 @@ Follow the tutorial below to test run the followup bot on your Zulip instance
 
   ```
   [api]
-  key=<api-key>
-  email=<email>
+  key=<api-key-of-the-bot>
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: I don't agree with this change - it's intentional to allow using the bot's or an existing user's credentials for this task.

File: tasks/interactive-bots.md [hunk 3]
```diff
@@ -54,19 +54,17 @@ Follow the tutorial below to test run the followup bot on your Zulip instance
...
+  email=<email-of-the-bot>
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: Same as [line 57](https://github.com/zulip/zulip-gci/pull/428/files#diff-fe4c8f75035983d1f4587df52529c417R57).

File: tasks/interactive-bots.md [hunk 4]
```diff
@@ -54,19 +54,17 @@ Follow the tutorial below to test run the followup bot on your Zulip instance
...
   site=<dev-url>
   ```
 
-  In the example replace the placeholder `<api-key>` with an existing API
-  key and placeholder `<email>` with the bot's email. The placeholder 
-  `<dev-url>` should point to your development environment URL.
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: Same as [line 57](https://github.com/zulip/zulip-gci/pull/428/files#diff-fe4c8f75035983d1f4587df52529c417R57).

File: tasks/interactive-bots.md [hunk 5]
```diff
@@ -80,7 +78,7 @@ Follow the tutorial below to test run the followup bot on your Zulip instance
 
   Make sure to point to your `.zuliprc-local` file - if you have created it in
   a different folder than `~`, you have to point to the right folder, e.g.
-  `../../.zuliprc-local`.
+  `<folder-name>/<folder-name>/.zuliprc-local`.
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: I don't agree with this change - this is an appropriate usage of the Terminal meaning "this file is two folders up".

File: tasks/interactive-bots.md [hunk 6]
```diff
@@ -94,14 +92,14 @@ by:
...
  run the bot and the output with the bot description
- - messages sent to the bot in the `devel` and `social` streams
- - messages sent by the bot in the followup stream
+ - messages sent by you to the bot in the `devel` and `social` streams
+ - messages sent by the bot in the `followup` stream
  - any other screenshots you find relevant
 
 Add the screenshots to `interactive-bots/followup/<username>/`. Make sure your
-filenames include no white spaces or underscores, use dashes (`-`) as connectors.
+filenames do not have spaces. Instead, use dashes (`-`).
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: It's unclear to me if this clarifies anything, especially since "white space" is not the same as "space" - the former is a more general term used often when developing. I don't think this one is necessary.

File: tasks/interactive-bots.md [hunk 7]
```diff
@@ -123,7 +121,7 @@ We recommend completing Task Type A before doing this task.
 Follow the tutorial below to create your first simple bot that sends private
 messages to users.
 
-* Make a copy of `followup.py` under `contrib_bots/lib` and name it `links.py`, to build on existing code.
+* Make a copy of `followup.py` located in `contrib_bots/lib` and name it `links.py`, to build on existing code.
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: I think `under` is more appropriate here.

File: tasks/interactive-bots.md [hunk 8]
```diff
@@ -203,21 +201,21 @@ messages to users.
...
+ - the output in your terminal window with the bot running, including the command you used to
  run the bot and the output with the bot description
- - messages sent to the bot in the `devel` and `social` streams
+ - messages sent by you to the bot in the `devel` and `social` streams
  - messages sent by the bot in the private messages to the author
  - any other screenshots you find relevant
 
  Add the screenshots to `interactive-bots/links/<username>/`. Make sure your
- filenames include no white spaces or underscores, use dashes (`-`) as connectors.
+ filenames do not have spaces. Instead, use dashes (`-`).
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: Same as [line 102].

File: tasks/interactive-bots.md [hunk 9]
```diff
@@ -263,16 +261,16 @@ requirements.
...
-should it react to, what should it do with the message content and where should
-it post any output.
+Make sure to make clear what would be expected of the bot - the kind of messages
+it should react to, what it should do with the message content and where it
+should post any output.
 
 * Present your idea for an interactive bot on the `interactive bots` topic on
 the `GCI tasks` stream. Consult with mentors and make sure your idea is approved
- before you start coding.
+ before you start working on it.
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: Communication is also working on the task. Coding is implementing a bot you've agreed on with a mentor. I think we can change to "implementing" but not "working".

File: tasks/interactive-bots.md [hunk 10]
```diff
@@ -284,7 +282,7 @@ the `GCI tasks` stream. Consult with mentors and make sure your idea is approved
   * The bot posts the expected output.
 
 * Document your interactive bot at `~/zulip/contrib_bots/lib/<bot_name>` in a
-`docs.md`. Add any relevant screenshots that will help understand how the bot
+`docs.md` (about the bot). Add any relevant screenshots that will help understand how the bot
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: I think this is a tad superfluous, since it's clear from the context. Does it really add clarity?

File: tasks/interactive-bots.md (hunk 5 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: OK I'll change that.

File: tasks/interactive-bots.md (hunk 9 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: OK

File: tasks/interactive-bots.md (hunk 10 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: I really didn't get what to document first.
So, I thought adding this text makes it more clear.
That's what I feel, but if you want, I can change it back.

File: tasks/interactive-bots.md (hunk 7 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: But don't you think "located in" is more... understandable?

File: tasks/interactive-bots.md (hunk 6 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: I've changed that.

File: tasks/interactive-bots.md (hunk 1 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: Line 46, of the un-edited or the edited file?

File: tasks/interactive-bots.md (hunk 2 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: OK, I'll restore that.

File: tasks/interactive-bots.md (hunk 4 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: I'll restore that too.

File: tasks/interactive-bots.md (hunk 8 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: Done.

File: tasks/interactive-bots.md [hunk 11]
```diff
@@ -59,14 +59,14 @@ Follow the tutorial below to test run the followup bot on your Zulip instance
   site=<dev-url>
   ```
 
-  In the example replace the placeholder `<api-key>` with an existing API
-  key and placeholder `<email>` with the bot's email. The placeholder 
-  `<dev-url>` should point to your development environment URL.
+-  In the example replace the placeholder `<api-key>` with an existing API
+-  key and placeholder `<email>` with the bot's email.The placeholder
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: Why did you remove the whitespaces here? There should be a space before and after `The placeholder`.

File: tasks/interactive-bots.md (hunk 7 above)
Comment (trueskawka) [inline tasks/interactive-bots.md]: I don't think so - `located in` suggest a file already exists, but it doesn't.

File: tasks/interactive-bots.md (hunk 10 above)
Comment (trueskawka) [inline tasks/interactive-bots.md]: So maybe add more clarity? E.g. `(description of how the bot works, example usage etc.)`?

File: tasks/interactive-bots.md (hunk 11 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: I'll get a trailing white-space error when I run `lint-all`

Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: Or is it not there in this?

File: tasks/interactive-bots.md (hunk 10 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: OK

File: tasks/interactive-bots.md [hunk 12]
```diff
@@ -59,14 +59,14 @@ Follow the tutorial below to test run the followup bot on your Zulip instance
   site=<dev-url>
   ```
 
-  In the example replace the placeholder `<api-key>` with an existing API
-  key and placeholder `<email>` with the bot's email. The placeholder 
-  `<dev-url>` should point to your development environment URL.
+-  In the example replace the placeholder `<api-key>` with an existing API
```
Comment (umairwaheed) [inline tasks/interactive-bots.md]: I think we can get rid of the word 'placeholder'.

File: tasks/interactive-bots.md [hunk 13]
```diff
@@ -59,14 +59,14 @@ Follow the tutorial below to test run the followup bot on your Zulip instance
...
+-  key and placeholder `<email>` with the bot's email. The placeholder 
+-  `<dev-url>` should point to your development environment URL.
 
   If you're running your bot in the droplet, alongside your Zulip instance, the
   `<dev-url>` will be `localhost:9991`.
 
-* Create a `followup` stream on your Zulip server.
+* Create a `followup` stream on your Zulip server (localhost).
```
Comment (umairwaheed) [inline tasks/interactive-bots.md]: Localhost doesn't apply to people who are using remote servers.

File: tasks/interactive-bots.md [hunk 14]
```diff
@@ -94,14 +94,14 @@ by:
 * Take screenshots showing that the bot is working, make sure to have screenshots
 of:
 
- - your terminal window with the bot running, including the command you used to
+ - the output in your terminal window with the bot running, including the command you used to
```
Comment (umairwaheed) [inline tasks/interactive-bots.md]: Hmm, I am not sure if this change is required.

File: tasks/interactive-bots.md [hunk 15]
```diff
@@ -203,21 +203,21 @@ messages to users.
...
+  - it reacts to messages starting with `@link` and `@resource`
 
-  - it sends a private message to the author of the original message
+  - it sends a private message to the sender of the original message
 
 * Take screenshots showing that the bot is working, make sure to have screenshots
 of:
 
- - your terminal window with the bot running, including the command you used to
+ - the output in your terminal window with the bot running, including the command you used to
```
Comment (umairwaheed) [inline tasks/interactive-bots.md]: This is not required as well in my opinion.

**Comment by umairwaheed** [comment]
@abhayKashyap03, left some feedback. Looking good so far.

File: tasks/interactive-bots.md (hunk 12 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: Removed

File: tasks/interactive-bots.md (hunk 13 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: Also removed

File: tasks/interactive-bots.md [hunk 16]
```diff
@@ -284,7 +284,7 @@ the `GCI tasks` stream. Consult with mentors and make sure your idea is approved
   * The bot posts the expected output.
 
 * Document your interactive bot at `~/zulip/contrib_bots/lib/<bot_name>` in a
-`docs.md`. Add any relevant screenshots that will help understand how the bot
+`docs.md` (about the bot, how the bot works, what the bot does, etc.). Add any relevant screenshots that will help understand how the bot
```
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: I've made the lines shorter

File: tasks/interactive-bots.md (hunk 3 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: Done

File: tasks/interactive-bots.md (hunk 7 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: This the `links bot` part, so `followup.py` is already there in `contrib_bots/lib`

File: tasks/interactive-bots.md (hunk 14 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: Removed

File: tasks/interactive-bots.md (hunk 15 above)
Comment (abhayKashyap03) [inline tasks/interactive-bots.md]: Removed this too

File: tasks/interactive-bots.md (hunk 1 above)
Comment (trueskawka) [inline tasks/interactive-bots.md]: I wanted to approve my review, but the brackets with the text are still on line 44.

@abhayKashyap03 could you take a look?

File: tasks/interactive-bots.md [hunk 17]
```diff
@@ -284,14 +285,14 @@ the `GCI tasks` stream. Consult with mentors and make sure your idea is approved
...
-works.
+`docs.md` (about the bot, how the bot works, what the bot does, etc.). Add any relevant
+ screenshots that will help understand how the bot works.
 
 * Create a commit with your bot, with commit message `interactive bots: Create
 <bot_name> bot.`.
 
 * Create a pull request in the `zulip/zulip` repository, with title
-`interactive bots: Create <bot_name> bot by <username>.`. Link to your GCI task in comment section.
+`interactive bots: Create <bot_name> bot by <username>.`. Link of your pull request to your GCI task in comment section.
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: @abhayKashyap03 @TomaszKolek this is not right - the student was supposed to link their GCI task in their PR.

File: tasks/interactive-bots.md [hunk 18]
```diff
@@ -256,19 +256,20 @@ should it post any output. Should you have any doubts, discuss them in the Zulip
 *feature* bot.`.
 
 * Create a pull request in the `zulip/zulip` repository, with title
-`interactive bots: Create *feature* bot.`. Link to your GCI task in comment section.
+`interactive bots: Create *feature* bot.`. Link of your pull request to
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: @abhayKashyap03 Same as [line 295](https://github.com/zulip/zulip-gci/pull/428/files#diff-fe4c8f75035983d1f4587df52529c417R295).

File: tasks/interactive-bots.md [hunk 19]
```diff
@@ -233,10 +233,10 @@ doing this setup process. Add your notes as a `notes.md` file to
 
 ### Task Type C: Create an interactive bot
 
-We recommend claiming one of these tasks after completing Task Type A
+We recommend claiming this task type after completing Task Type A
 and Task Type B.
 
-Let *feature* be the feature mentioned in the task that brought you here.
+Let **feature** be the feature mentioned in the task that brought you here.
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: @abhayKashyap03 @TomaszKolek This is also not right, since we're using `*feature*` later in the document.

File: tasks/interactive-bots.md [hunk 20]
```diff
@@ -203,21 +203,21 @@ messages to users.
...
  - your terminal window with the bot running, including the command you used to
  run the bot and the output with the bot description
- - messages sent to the bot in the `devel` and `social` streams
+ - messages sent by you to the bot in the `devel` and `social` streams
  - messages sent by the bot in the private messages to the author
  - any other screenshots you find relevant
 
  Add the screenshots to `interactive-bots/links/<username>/`. Make sure your
- filenames include no white spaces or underscores, use dashes (`-`) as connectors.
+ filenames do not have white-spaces. Instead, use dashes (`-`).
```
Comment (trueskawka) [inline tasks/interactive-bots.md]: @abhayKashyap03 This is `white space`, not `white-space`, you can check [here](https://www.wikiwand.com/en/Whitespace_character).

**Comment by trueskawka** [comment]
@abhayKashyap03 @TomaszKolek @umairwaheed  FYI

I've found a few changes that haven't been there in the earlier version I've seen and are not right. Going to fix them manually on master.
